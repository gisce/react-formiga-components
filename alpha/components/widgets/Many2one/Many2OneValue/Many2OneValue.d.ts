import { Many2OneValueProps } from "./Many2OneValue.types";
export declare const Many2OneValue: (props: Many2OneValueProps) => import("react/jsx-runtime").JSX.Element;
//# sourceMappingURL=Many2OneValue.d.ts.map