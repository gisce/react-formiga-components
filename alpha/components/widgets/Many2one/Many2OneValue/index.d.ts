export * from "./Many2OneValue";
//# sourceMappingURL=index.d.ts.map