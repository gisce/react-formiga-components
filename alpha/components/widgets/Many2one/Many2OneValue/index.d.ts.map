{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../../src/components/widgets/Many2one/Many2OneValue/index.ts"],"names":[],"mappings":"AAAA,cAAc,iBAAiB,CAAA"}