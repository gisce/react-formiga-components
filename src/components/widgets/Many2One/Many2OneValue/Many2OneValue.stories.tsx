import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { action } from "@storybook/addon-actions";

import { Many2OneValue } from "./Many2OneValue";
import { Many2OneSuffix } from "../Many2OneSuffix/Many2OneSuffix";
import { Menu } from "antd";
import {
  PrinterOutlined,
  ThunderboltOutlined,
  EnterOutlined,
} from "@ant-design/icons";

export default {
  title: "Work in progress/Widgets/Many2One/Many2OneValue",
  component: Many2OneValue,
} as ComponentMeta<typeof Many2OneValue>;

const Template: ComponentStory<typeof Many2OneValue> = (args) => {
  return <Many2OneValue {...args} />;
};

export const Basic = Template.bind({});
Basic.args = {
  value: "Customer #23",
};

export const BasicWithSuffix = Template.bind({});
BasicWithSuffix.args = {
  value: "Customer #23",
  // suffix: (
  //   <Many2OneSuffix
  //     loading={false}
  //     menuItems={
  //       <>
  //         <Menu.Item key="action_1" icon={<ThunderboltOutlined />}>
  //           action_1
  //         </Menu.Item>
  //         <Menu.Item key="action_2" icon={<PrinterOutlined />}>
  //           action_2
  //         </Menu.Item>
  //         <Menu.Item key="action_3" icon={<EnterOutlined />}>
  //           action_3
  //         </Menu.Item>
  //       </>
  //     }
  //     onMenuItemClicked={(e) => {
  //       action("onMenuItemClicked")(e);
  //     }}
  //     onSuffixExpandClicked={() => {
  //       action("onSuffixExpandClicked")();
  //     }}
  //   />
  // ),
};
