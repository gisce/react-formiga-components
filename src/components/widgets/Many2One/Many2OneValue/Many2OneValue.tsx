import { Space } from "antd";
import React from "react";
import { Many2OneValueProps } from "./Many2OneValue.types";

export const Many2OneValue = (props: Many2OneValueProps) => {
  const { value, suffix = null } = props;
  return (
    <Space>
      <>{value}</>
      {suffix}
    </Space>
  );
};
