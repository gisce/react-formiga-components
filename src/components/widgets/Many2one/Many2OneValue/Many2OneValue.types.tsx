export type Many2OneValueProps = {
  value: string;
  suffix: React.ReactNode;
};
