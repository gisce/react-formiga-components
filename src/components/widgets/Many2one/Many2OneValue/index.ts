export * from "./Many2OneValue"