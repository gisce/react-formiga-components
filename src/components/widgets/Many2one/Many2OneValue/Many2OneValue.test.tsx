import React from "react";
import { render } from "@testing-library/react";
import { composeStories } from "@storybook/testing-react";
import * as stories from "./Many2OneValue.stories";
import matchMediaMock from "mocks/matchMedia.mock";

describe("Many2OneValue", () => {
  beforeAll(() => {
    matchMediaMock();
  });

  const testCases = Object.values(composeStories(stories)).map((Story) => [
    Story.storyName!,
    Story,
  ]);

  test.each(testCases)("Renders %s story", async (_storyName, Story) => {
    const tree = render(<Story />);
    expect(tree.baseElement).toMatchSnapshot();
  });
});
