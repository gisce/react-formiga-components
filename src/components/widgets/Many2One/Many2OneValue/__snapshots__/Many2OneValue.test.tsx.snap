// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Many2OneValue Renders Basic story 1`] = `
<body>
  <div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
      >
        Customer #23
      </div>
    </div>
  </div>
</body>
`;

exports[`Many2OneValue Renders BasicWithSuffix story 1`] = `
<body>
  <div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        Customer #23
      </div>
      <div
        class="ant-space-item"
      >
        <span
          aria-label="right-circle"
          class="anticon anticon-right-circle ant-dropdown-trigger"
          role="img"
          style="color: rgba(0, 0, 0, 0.45);"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="right-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M666.7 505.5l-246-178A8 8 0 00408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
            />
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</body>
`;
